test/embind/test_i64_val.cpp

```diff
@@ -43,33 +57,16 @@ int main() {
   printf("start\n");
   EM_ASM({globalThis.a = null});
 
-  test("val(int64_t v)");
-  val::global().set("a", val(int64_t(1234)));
-  ensure_js("a === 1234n");
-
-  val::global().set("a", val(int64_t(-4321)));
-  ensure_js("a === -4321n");
-
-  val::global().set("a", val(int64_t(0x12345678aabbccddL)));
-  ensure_js("a === 1311768467732155613n");
-  assert(val::global()["a"].as<int64_t>() == 0x12345678aabbccddL);
-
   test("val(uint64_t v)");
-  val::global().set("a", val(uint64_t(1234)));
-  ensure_js("a === 1234n");
-
-  val::global().set("a", val(max_uint64_t));
-  ensure_js(compare_a_64_js(max_uint64_t));
-  assert(val::global()["a"].as<uint64_t>() == max_uint64_t);
+  test_value(uint64_t(1234));
+  test_value(max_uint64_t);
 
   test("val(int64_t v)");
-  val::global().set("a", val(max_int64_t));
-  ensure_js(compare_a_64_js(max_int64_t));
-  assert(val::global()["a"].as<int64_t>() == max_int64_t);
-
-  val::global().set("a", val(min_int64_t));
-  ensure_js(compare_a_64_js(min_int64_t));
-  assert(val::global()["a"].as<int64_t>() == min_int64_t);
+  test_value(int64_t(1234));
+  test_value(int64_t(-4321));
+  test_value(int64_t(0x12345678aabbccddL));
+  test_value(min_int64_t);
+  test_value(max_int64_t);
```

Comment:
Does this test change need to happen as part of this change?

What I'm kind of aksing here is, does the unmodified/old test no longer pass/work after this change?   Maybe this test improvement can land separately / first?

Reply:
At to be clear the old test prior to this PR still passes with the changes in this PR?

Reply:
Yeah, it just got extended to also test equivalent of `BigInt(value)` and `globalThis.BigInt(value)` in addition to pre-existing `a = value;`.